test/zeroD/test_zeroD.cpp

```diff
@@ -50,6 +52,185 @@ TEST(ZeroDim, test_individual_reactor_initialization)
     }
 }
 
+TEST(ZeroDim, test_get_reaction_from_reactor)
+{
+    // Setting up solution to insert in reactor
+    auto sol = newSolution("methane_twostep.yaml");
+    sol->thermo()->setState_TPX(300.0, 101325, "CH4:1.0, O2:1.0, H2O:0, CO2:0");
+    // Set up reactor object
+    Reactor reactor;
+    reactor.insert(sol);
+    // Get reactions
+    auto kinetics = reactor.getKineticsMgr();
+    auto reaction0 = kinetics->reaction(0);
+    auto reaction1 = kinetics->reaction(1);
+    // Compare reaction strings to manual input
+    EXPECT_EQ(reaction0->equation(), "CH4 + 1.5 O2 => CO + 2 H2O");
+    EXPECT_EQ(reaction1->equation(), "CO + 0.5 O2 <=> CO2");
+}
+
+TEST(MoleReactorTestSet, test_preconditioned_ideal_const_vol)
+{
+    // Reactor 1
+    auto sol1 = newSolution("h2o2.yaml");
+    sol1->thermo()->setState_TP(1100.0, 10 * OneAtm);
+    sol1->thermo()->setEquivalenceRatio(1, "H2", "O2:0.21, N2:0.78, AR:0.09");
+    IdealGasMoleReactor r1;
+    r1.insert(sol1);
+    // Reactor 2
+    auto sol2 = newSolution("h2o2.yaml");
+    sol2->thermo()->setState_TP(1100.0, 10 * OneAtm);
+    sol2->thermo()->setEquivalenceRatio(1, "H2", "O2:0.21, N2:0.78, AR:0.09");
+    IdealGasMoleReactor r2;
+    r2.insert(sol2);
+    // Network 1
+    ReactorNet net1;
+    net1.addReactor(r1);
+    AdaptivePreconditioner precon;
+    net1.setProblemType(GMRES);
+    net1.setPreconditioner(precon);
+    net1.initialize();
+    // Network 2
+    ReactorNet net2;
+    net2.addReactor(r2);
+    net2.setProblemType(GMRES);
+    net2.initialize();
+    // Advancing
+    net1.advance(1.0);
+    net2.advance(1.0);
+    // Comparison
+    vector_fp state1(r1.neq(), 0.0);
+    vector_fp state2(r2.neq(), 0.0);
+    for (size_t i = 0; i < r1.neq(); i++)
+    {
+        EXPECT_NEAR(state1[i], state2[i], net1.atol());
+    }
+}
+
+TEST(MoleReactorTestSet, test_preconditioned_ideal_const_pressure)
+{
+    // Reactor 1
+    auto sol1 = newSolution("h2o2.yaml");
+    sol1->thermo()->setState_TP(1100.0, 10 * OneAtm);
+    sol1->thermo()->setEquivalenceRatio(1, "H2", "O2:0.21, N2:0.78, AR:0.09");
+    IdealGasConstPressureMoleReactor r1;
+    r1.insert(sol1);
+    // Reactor 2
+    auto sol2 = newSolution("h2o2.yaml");
+    sol2->thermo()->setState_TP(1100.0, 10 * OneAtm);
+    sol2->thermo()->setEquivalenceRatio(1, "H2", "O2:0.21, N2:0.78, AR:0.09");
+    IdealGasConstPressureMoleReactor r2;
+    r2.insert(sol2);
+    // Network 1
+    ReactorNet net1;
+    net1.addReactor(r1);
+    AdaptivePreconditioner precon;
+    net1.setProblemType(GMRES);
+    net1.setPreconditioner(precon);
+    // Network 2
+    ReactorNet net2;
+    net2.addReactor(r2);
+    net2.setProblemType(GMRES);
+    net2.initialize();
+    // Advancing
+    net1.advance(1.0);
+    net2.advance(1.0);
+    // Comparison
+    vector_fp state1(r1.neq(), 0.0);
+    vector_fp state2(r2.neq(), 0.0);
+    for (size_t i = 0; i < r1.neq(); i++)
+    {
+        EXPECT_NEAR(state1[i], state2[i], net1.atol());
+    }
```

Comment:
`state1` and `state2` are just vectors of zeros here - I think you forgot to at least get the state of the two reactors.